windows-gaming-driver/src/util.rs

```diff
@@ -23,3 +23,23 @@ pub fn usable_ports(bus: UsbBus) -> usize {
     }
 }
 
+pub trait PrettySplit<T>{
+	fn into_two(self, &Fn(&T) -> (bool)) -> (Vec<T>, Vec<T>);
```

Comment:
This already exists. It's called [partition](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition).